rocketpy/control/controller.py

```diff
@@ -2,42 +2,58 @@
+    the objects that are passed to it. The controller function also stores the
+    variables of interest in the objects that are passed to it."""
 
     def __init__(
         self,
-        observed_objects,
+        interactable_objects,
         controller_function,
         sampling_rate,
+        initial_observed_variables=None,
         name="Controller",
     ):
         """Initialize the class with the controller function and the objects to
         be observed.
 
         Parameters
         ----------
-        observed_objects : list
-            A list of objects to be observed by the controller. It can be any
-            python object. This list will be passed to the controller function
-            as positional arguments, meaning that the order of the objects in
-            this list matters. These objects will then be able to be accessed
-            and modified by the controller function.
+        interactable_objects : list
+            A collection of objects that the controller function can access and
+            potentially modify. This list is flexible and can include any Python
+            object. The objects listed here are provided to the controller
+            function as the last argument, maintaining the order specified in
+            this list. The controller function gains the ability to interact
+            with and make adjustments to these objects during its execution.
         controller_function : function, callable
-            A function that takes the following arguments, in this order:
-
-            1. Time of the simulation at the current step in seconds.
-            2. The sampling rate at which the controller function is called in
-               Hertz (Hz).
-            3. The state vector of the simulation, which is defined as:
+            An user-defined function responsible for controlling the simulation.
+            This function is expected to take the following arguments, in order:
 
+            1. `time` (float): The current simulation time in seconds.
+            2. `sampling_rate` (float): The rate at which the controller
+               function is called, measured in Hertz (Hz).
+            3. `state` (list): The state vector of the simulation, structured as
                `[x, y, z, vx, vy, vz, e0, e1, e2, e3, wx, wy, wz]`.
-            4. A list containing the state history of the simulation. The state
-               history is a list of every state vector of every step of the
-               simulation. The state history is a list of lists, where each
-               sublist is a state vector and is ordered from oldest to newest.
-            5. A list containing the objects to be acted upon by the controller.
-               The objects in this list are the same as the objects in the
-               observed_objects list, but they can be modified by the controller.
+            4. `state_history` (list): A record of the rocket's state at each
```

Comment:
so the `state_history` works like the same as the `Flight.solution` attribute, right?
I wonder if each state could/should be an instance of a class, so we would be saving a list of objects of such class, instead of saving a list of lists.

Reply:
It is an interesting idea. Maybe being a named tuple would be more fitting

It is something to consider whenever we work on some refactoring in the Flight class